crates/ruma-signatures/src/error.rs

```diff
@@ -41,11 +43,25 @@ pub enum Error {
     PduSize,
 }
 
+impl From<RedactionError> for Error {
+    fn from(err: RedactionError) -> Self {
+        match err {
+            RedactionError::NotOfType { field: target, of_type, .. } => {
+                JsonError::NotOfType { target, of_type }.into()
+            }
+            RedactionError::JsonFieldMissingFromObject(field) => {
+                JsonError::JsonFieldMissingFromObject(field).into()
+            }
+            _ => unreachable!(),
```

Comment:
If we add another variant to `RedactionError`, does this trigger a cargo/clippy warning?

Reply:
Not right now, would require enabling a nightly-feature to be able to activate the right lint. I'm not expecting this to stay in long though, ruma-signatures' error types could use some refactoring and I think `RedactionError` should just become its own variant with that.

Reply:
Opened an issue: #1216